frontend/src/components/Register.tsx

```diff
@@ -17,11 +18,26 @@ const Register = () => {
     email: "",
     password: "",
   });
+  const [passwordStrength, setPasswordStrength] = useState<string>("");
+  const [passwordError, setPasswordError] = useState<string>("");
+  const handlePasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const password = event.target.value;
+    setAuthInputs({ ...authInputs, password });
+
+    const errors = validatePassword(password);
+    if (errors.length > 0) {
+      setPasswordError(errors.join(" "));
+      setPasswordStrength("Weak");
+    } else {
+      setPasswordError("");
+      setPasswordStrength("Strong");
+    }
+  };
```

Comment:
It would be great if you could create a custom hook usePassword and abstract this code inside the hook. 
The hook can return [passwordError, passwordStrength,.. other things that are needed]

The hook can be reused in both, Login & Register screens. Much more maintainable code.

Reply:
Password validations will not be used in Login page as they have created already during Signup process so adding these validations in Login might not be a good one I guess.

Reply:
Got it. In that case, this is not required then.